src/DependencyGraph.java

```diff
@@ -0,0 +1,169 @@
+import java.util.*;
+
+public class DependencyGraph {
+    private Map<DependencyVertex, List<DependencyVertex>> deps = new HashMap<>();
```

Comment:
Just a flag to think a bit about if this is going to be seen and/or mutated from multiple threads. A few solutions like `ConcurrentHashMap` and `Collections.synchronizedMap` exist for the "simple" case of just a bunch of maps thrashing, but the `List` in the Map is also mutable so...food for thought.